components/script/dom/htmlcanvaselement.rs

```diff
@@ -425,16 +424,17 @@ impl HTMLCanvasElementMethods for HTMLCanvasElement {
         };
 
         // FIXME: Only handle image/png for now.
-        let mut png = Vec::new();
+        let mut url = "data:image/png;base64,".to_owned();
+        let mut encoder = base64::write::EncoderStringWriter::from_consumer(
+            &mut url,
+            &base64::engine::general_purpose::STANDARD,
+        );
         // FIXME(nox): https://github.com/image-rs/image-png/issues/86
         // FIXME(nox): https://github.com/image-rs/image-png/issues/87
-        PngEncoder::new(&mut png)
+        PngEncoder::new(&mut encoder)
             .write_image(&file, self.Width(), self.Height(), ColorType::Rgba8)
             .unwrap();
-        let mut url = "data:image/png;base64,".to_owned();
-        // FIXME(nox): Should this use base64::URL_SAFE?
-        // FIXME(nox): https://github.com/marshallpierce/rust-base64/pull/56
-        base64::engine::general_purpose::STANDARD.encode_string(&png, &mut url);
+        encoder.into_inner();
```

Comment:
What's the purpose of this into_inner call if the return value is ignored?

Reply:
Its purpose is to poke the encoder into finishing encoding everything while at the same time dropping the `&mut url` reference that the encoder had borrowed; So yeah, removing this call isn't possible, and plus, it can't be directly given to `USVString` either without converting the reference into an owned value, which I believe would effectively duplicate the `url` variable?

Do note that the way the encoding is done matches the one described on `base64`'s documentation, so I believe this is how it's intended to be used